Tests/UnitTests/Purchasing/Purchases/BasePurchasesTests.swift

```diff
@@ -113,12 +113,14 @@ class BasePurchasesTests: TestCase {
             Purchases.clearSingleton()
             self.clearReferences()
 
-            expect(purchases)
-                .toEventually(beNil(), description: "Purchases has leaked")
-            expect(orchestrator)
-                .toEventually(beNil(), description: "PurchasesOrchestrator has leaked")
-            expect(deviceCache)
-                .toEventually(beNil(), description: "DeviceCache has leaked: \(self)")
+            // Note: this captures the boolean to avoid race conditions when Nimble tries
+            // to print the instances while they're being deallocated.
```

Comment:
didn't know it was possible to capture this as a boolean, I thought this block would capture strongly instead. How does this work? like, doesn't it need the object in order to reference when re-evaluating?

Reply:
The references are `weak` though (defined above). The previous implementation used an `@autoclosure`, so `expect(purchases)` was converted to `expect { purchases }`.
Internally it was using reflection (possibly `String.init(describing:)`) to print a value for that object.

Now when it generates a description from the subject of the expectation, it's just a boolean.

Let me know if that makes sense.